### PR TITLE
Allow configuring allowed CORS origins for suggest

### PR DESCRIPTION
**Pull request by paulmelnikow**
- Support single-server testing and a local dev server (#1273) that is on a different port from the shields server
- Refactor config schema

With this change, the suggestions work locally in #1273.

### CODE REVIEW

**Comment by paulmelnikow** [comment]
Companion PR: badges/ServerScript#2

**Comment by platan** [comment]
Code looks good for me. I'm going to run it and test it tomorrow.

File: doc/self-hosting.md [hunk 1]
```diff
@@ -139,8 +139,17 @@ If you want to host the frontend on a separate server, such as cloud storage
...
 To help out users, you can make the Shields server redirect the server root.
-Set the `FRONTEND_REDIRECT_URL` environment variable:
+Set the `REDIRECT_URI` environment variable:
 
 ```sh
-FRONTEND_REDIRECT_URL=http://my-custom-shields.s3.amazonaws.com/
+REDIRECT_URI=http://my-custom-shields.s3.amazonaws.com/
 ```
+
+If you want to use server suggestions, you should also set `ALLOWED_ORIGIN`:
```
Comment (platan) [inline doc/self-hosting.md]: In line 20 in this file, there is an info that one should build frontend using `BASE_URL=https://your-server.example.com npm run build:production`
`build:production` runs `make footer-production-transform`, which sets `img.shields.io` as a suggestions server (`var origin = 'https://img.shields.io';` in `index.html`). Requests from locally build frontend will go to img.shields.io.

Comment (paulmelnikow) [reply doc/self-hosting.md]: Thanks for noticing that. It should be fixed as part of #1273, which gets rid of the makefile transform and ships the suggestions server through the bundle.

**Comment by paulmelnikow** [comment]
Thanks for the careful read + review!

**Pull request merged**